Extensions/AzureAdAuthenticationBuilderExtensions.cs

```diff
@@ -19,28 +21,41 @@ public static AuthenticationBuilder AddAzureAd(this AuthenticationBuilder builde
             return builder;
         }
 
-        private class ConfigureAzureOptions: IConfigureNamedOptions<OpenIdConnectOptions>
+        private class ConfigureAzureOptions : IConfigureNamedOptions<OpenIdConnectOptions>
         {
             private readonly AzureAdOptions _azureOptions;
+            private readonly ITokenAcquisition _tokenAcquisition;
 
-            public ConfigureAzureOptions(IOptions<AzureAdOptions> azureOptions)
+            public ConfigureAzureOptions(IOptions<AzureAdOptions> azureOptions, ITokenAcquisition tokenAcquisition)
             {
                 _azureOptions = azureOptions.Value;
+                _tokenAcquisition = tokenAcquisition;
             }
 
             public void Configure(string name, OpenIdConnectOptions options)
             {
                 options.ClientId = _azureOptions.ClientId;
-                options.Authority = $"{_azureOptions.Instance}common/v2.0";   // V2 specific
+                options.Authority = $"{_azureOptions.Instance}{_azureOptions.TenantId}/v2.0";   // V2 specific
                 options.UseTokenLifetime = true;
                 options.RequireHttpsMetadata = false;
                 options.TokenValidationParameters.ValidateIssuer = false;     // accept several tenants
+                options.Events = new OpenIdConnectEvents();
+                options.Events.OnAuthorizationCodeReceived = OnAuthorizationCodeReceived;
+                options.Scope.Add("user.read");
+
+                options.ResponseType = "code id_token";
             }
 
             public void Configure(OpenIdConnectOptions options)
             {
                 Configure(Options.DefaultName, options);
             }
+
+            private async Task OnAuthorizationCodeReceived(AuthorizationCodeReceivedContext context)
+            {
+                string[] scopes = new string[] { "user.read" };
```

Comment:
no need for local variable 'scopes'.
async will need to make a copy, just pass in 'new string[] { "user.read" }
OR define a constant.

Reply:
Good point.

Reply:
I would probably have used a const, as that improves readability

---
In reply to: [215816760](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/7#discussion_r215816760) [](ancestors = 215816760)